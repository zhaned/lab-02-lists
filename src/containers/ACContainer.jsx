import React, { Component } from 'react';
import VillagerList from '../components/villagers/VillagerList';
import { findVillagers } from '../services/animalCrossingApi';

export default class ACContainer extends Component {
  state = {
    loading: true,
    villagers: []
  }

  async componentDidMount() {
    const villagers = await findVillagers();
    this.setState({
      villagers,
      loading: false
    });
  }

  render() {
    const { loading, villagers } = this.state;

    if(loading) return <h1>Gathering Villagers...</h1>
    return <VillagerList villagers={villagers} />
  }
}