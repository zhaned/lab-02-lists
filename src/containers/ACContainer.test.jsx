import React from 'react';
import { render, screen } from '@testing-library/react';
import ACContainer from './ACContainer';

describe('ACContainer', () => {
  it('renders page of villagers', async () => {
    render(<ACContainer />);

    const unorderedList = await screen.findByRole('list', { name: 'villagers' });
    expect(unorderedList).toMatchSnapshot();
  });
});