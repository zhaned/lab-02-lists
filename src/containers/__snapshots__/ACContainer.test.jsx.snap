// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACContainer renders page of villagers 1`] = `
<ul
  aria-label="villagers"
>
  <li>
    <img
      alt="Admiral"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
    />
    <p>
      Admiral
    </p>
    <p>
      Only quitters give up.
    </p>
  </li>
  <li>
    <img
      alt="Agent S"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
    />
    <p>
      Agent S
    </p>
    <p>
      You gotta put the pedal to the metal!
    </p>
  </li>
  <li>
    <img
      alt="Agnes"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
    />
    <p>
      Agnes
    </p>
    <p>
      You reap what you sow.
    </p>
  </li>
  <li>
    <img
      alt="Al"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Al_NH.png/revision/latest?cb=20200802150354"
    />
    <p>
      Al
    </p>
    <p>
      Life is a buffet--Always go back for seconds.
    </p>
  </li>
  <li>
    <img
      alt="Alfonso"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e4/Alfonso_NH.png/revision/latest?cb=20200817164717"
    />
    <p>
      Alfonso
    </p>
    <p>
      Slow and steady wins the race.
    </p>
  </li>
  <li>
    <img
      alt="Alice"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e1/Alice_NH.png/revision/latest?cb=20200803181400"
    />
    <p>
      Alice
    </p>
    <p>
      Treat others as you would like to be treated.
    </p>
  </li>
  <li>
    <img
      alt="Ankha"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/56/Ankha_NH.png/revision/latest?cb=20200817172044"
    />
    <p>
      Ankha
    </p>
    <p>
      All that glitters is not gold.
    </p>
  </li>
  <li>
    <img
      alt="Alli"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/8e/Alli_NH.png/revision/latest?cb=20200802061731"
    />
    <p>
      Alli
    </p>
    <p>
      Think before you act.
    </p>
  </li>
  <li>
    <img
      alt="Anicotti"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
    />
    <p>
      Anicotti
    </p>
    <p>
      The walls have ears!
    </p>
  </li>
  <li>
    <img
      alt="Angus"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/5f/Angus_NH.png/revision/latest?cb=20200802083758"
    />
    <p>
      Angus
    </p>
    <p>
      Even strong men cry.
    </p>
  </li>
  <li>
    <img
      alt="Amelia"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/ab/Amelia_NH.png/revision/latest?cb=20200817171545"
    />
    <p>
      Amelia
    </p>
    <p>
      Pampering yourself is as important as food, water, and shelter!
    </p>
  </li>
  <li>
    <img
      alt="Annalisa"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/f/f0/Annalisa_NH.png/revision/latest?cb=20200802063308"
    />
    <p>
      Annalisa
    </p>
    <p>
      Home is where the heart is.
    </p>
  </li>
  <li>
    <img
      alt="Antonio"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d4/Antonio_NH.png/revision/latest?cb=20200802063352"
    />
    <p>
      Antonio
    </p>
    <p>
      Always go for the burn!
    </p>
  </li>
  <li>
    <img
      alt="Annalise"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
    />
    <p>
      Annalise
    </p>
    <p>
      Let gossip in one ear and out the other.
    </p>
  </li>
  <li>
    <img
      alt="Anabelle"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d8/Anabelle_NH.png/revision/latest?cb=20200802063200"
    />
    <p>
      Anabelle
    </p>
    <p>
      Dance like no one's watching!
    </p>
  </li>
  <li>
    <img
      alt="Apollo"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/2/28/Apollo_NH.png/revision/latest?cb=20200719092927"
    />
    <p>
      Apollo
    </p>
    <p>
      What goes up must come down.
    </p>
  </li>
  <li>
    <img
      alt="Anchovy"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ea/Anchovy_NH.png/revision/latest?cb=20200802081231"
    />
    <p>
      Anchovy
    </p>
    <p>
      Why do today what you can put off until tomorrow?
    </p>
  </li>
  <li>
    <img
      alt="Apple"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/57/Apple_NH.png/revision/latest?cb=20200802151508"
    />
    <p>
      Apple
    </p>
    <p>
      One rotten apple spoils the barrel!
    </p>
  </li>
  <li>
    <img
      alt="Astrid"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954"
    />
    <p>
      Astrid
    </p>
    <p>
      An apple doesn't fall far from the tree.
    </p>
  </li>
  <li>
    <img
      alt="Audie"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/1/1b/Audie_NH.png/revision/latest?cb=20200829012735"
    />
    <p>
      Audie
    </p>
    <p>
      Be the kind of person your future self won't regret having been
    </p>
  </li>
  <li>
    <img
      alt="Aurora"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/59/Aurora_NH.png/revision/latest?cb=20200202215217"
    />
    <p>
      Aurora
    </p>
    <p>
      Always keep your cool.
    </p>
  </li>
  <li>
    <img
      alt="Ava"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/cc/Ava_NH.png/revision/latest?cb=20200609055741"
    />
    <p>
      Ava
    </p>
    <p>
      Early to bed, early to rise.
    </p>
  </li>
  <li>
    <img
      alt="Avery"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/dd/Avery_NH.png/revision/latest?cb=20200608222354"
    />
    <p>
      Avery
    </p>
    <p>
      What goes around comes around.
    </p>
  </li>
  <li>
    <img
      alt="Baabara"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/df/Baabara_NH.png/revision/latest?cb=20200802165552"
    />
    <p>
      Baabara
    </p>
    <p>
      Don't just follow the flock.
    </p>
  </li>
  <li>
    <img
      alt="Barold"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/9/9d/Barold_NH.png/revision/latest?cb=20200802141140"
    />
    <p>
      Barold
    </p>
    <p>
      Please DO feed the bears!
    </p>
  </li>
</ul>
`;
